src/adapter/src/coord/appends.rs

```diff
@@ -229,18 +268,23 @@ impl<S: Append + 'static> Coordinator<S> {
             //      empty writes, etc) which will drop the lock. The deferred group
             //      commit will then acquire the lock and execute a group commit.
             self.defer_write(Deferred::GroupCommit);
-        }
-    }
 
-    /// Commits all pending write transactions at the same timestamp. All pending writes will be
-    /// combined into a single Append command and sent to STORAGE as a single batch. All writes will
-    /// happen at the same timestamp and all involved tables will be advanced to some timestamp
-    /// larger than the timestamp of the write.
-    #[tracing::instrument(level = "debug", skip_all)]
-    pub(crate) async fn group_commit(&mut self) {
-        if self.pending_writes.is_empty() {
-            return;
-        }
+            // Without the write lock we can only apply writes to system tables.
+            // TODO(jkosh44) replace with drain_filter when it's stable.
+            let mut pending_writes = Vec::new();
+            let mut i = 0;
+            while i < self.pending_writes.len() {
+                if matches!(&self.pending_writes[i], PendingWriteTxn::System(_)) {
+                    pending_writes.push(self.pending_writes.remove(i));
```

Comment:
I think `swap_remove` seems like a good idea. We don't care about the order because everything gets shoved into a `HashMap` which destroys the order anyways. Updated to use `swap_remove`.